sdk/search/azure-search-documents/src/main/java/com/azure/search/documents/indexes/FieldBuilder.java

```diff
@@ -38,28 +43,26 @@
         SUPPORTED_NONE_PARAMETERIZED_TYPE.put(String.class, SearchFieldDataType.STRING);
         SUPPORTED_NONE_PARAMETERIZED_TYPE.put(Date.class, SearchFieldDataType.DATE_TIME_OFFSET);
         SUPPORTED_NONE_PARAMETERIZED_TYPE.put(OffsetDateTime.class, SearchFieldDataType.DATE_TIME_OFFSET);
-//        SUPPORTED_NONE_PARAMETERIZED_TYPE.put(PointGeometry.class, SearchFieldDataType.GEOGRAPHY_POINT);
+        SUPPORTED_NONE_PARAMETERIZED_TYPE.put(PointGeometry.class, SearchFieldDataType.GEOGRAPHY_POINT);
     }
 
-    private static final List<Class<?>> UNSUPPORTED_TYPES = Arrays.asList(Byte.class,
-        CharSequence.class,
-        Character.class,
-        char.class,
-        Float.class,
-        float.class,
-        Short.class,
-        short.class);
+    private static final List<Class<?>> UNSUPPORTED_TYPES = Arrays.asList(byte.class, Byte.class,
+        CharSequence.class, Character.class, char.class,
+        Float.class, float.class,
+        Short.class, short.class);
```

Comment:
Would we be allowed to upcast these types into `Double` and `Integer` respectively? Possible the same with `Byte` and `byte` to `Integer`?

Or is the idea being more restrictive is safer?

Reply:
There is no hard requirement on this currently. 
The reasons why we restrict more are mainly:
1. Adding more to allowlist is allowed for the release, but taking back is kind of breaking changes. I prefer to make it as limited as possible to minimize breaking changes we might introduce.
2. Mapping byte to integer is not quite right when serialize model POJO to Json. SearchField will use EDM.Integer, but upload doc would have it transfer to encoded string when send to the wire. It is really hard to predict how customer handle it. It is better to block it unitl we have clear use case.